File: Ahorn/entities/CorruptionCrystal.jl
module CorruptionCrystal

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/CorruptionCrystal" CorruptionCrystal(x::Integer, y::Integer, corruptionRadius::Integer=100, health::Integer=3, spreadSpeed::Integer=20)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(CorruptionCrystal)
)

function Ahorn.selection(entity::CorruptionCrystal)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::CorruptionCrystal, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/entities/corruptionCrystal.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local corruptionCrystal = {}

corruptionCrystal.name = "MaggyHelper/CorruptionCrystal"
corruptionCrystal.depth = -50
corruptionCrystal.fieldInformation = {
    health = {
        minimumValue = 1
    },
    corruptionRadius = {
        minimumValue = 0
    },
    spreadSpeed = {
        minimumValue = 0
    }
}
corruptionCrystal.placements = {
    {
        name = "normal",
        data = {
            health = 3,
            corruptionRadius = 100,
            spreadSpeed = 20
        }
    }
}

function corruptionCrystal.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("objects/corruption_crystal/dormant", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    sprite.color = {0.4, 0.0, 0.0, 1.0}
    return sprite
end

function corruptionCrystal.selection(room, entity)
    return utils.rectangle(entity.x - 16, entity.y - 48, 32, 48)
end

return corruptionCrystal
